src/window.cpp

```diff
@@ -2903,19 +2904,28 @@ static void MouseLoop(MouseClick click, int mousewheel)
 			case MC_LEFT:
 			case MC_DOUBLE_LEFT:
 				DispatchLeftClickEvent(w, x - w->left, y - w->top, click == MC_DOUBLE_LEFT ? 2 : 1);
-				break;
+				return;
 
 			default:
 				if (!scrollwheel_scrolling || w == NULL || w->window_class != WC_SMALLMAP) break;
 				/* We try to use the scrollwheel to scroll since we didn't touch any of the buttons.
 				 * Simulate a right button click so we can get started. */
 				FALLTHROUGH;
```

Comment:
Looks like scrollwheel scrolling doesn't simulate a right click any more (good!) in which case none of this default case should be necessary?

Reply:
Not quite true: it doesn't simulate right click in a viewport, but does still in any other window. So this default case is still necessary. I could separate scrollwheel scrolling for generic window handling just as I've done for viewport handling.

Reply:
I think not for now. In the future I'd like to see the event handling reworked including handling scrolling events properly, but for a fix this is fine.

Reply:
Yes, as I remarked in #6753, there's also some unused logic. So it could use some cleaning up anyways.